### PR TITLE
KC-1398: Implement controlled shutdown on the broker side

### PR DESCRIPTION
**Pull request by aloknnikhil**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java [hunk 1]
```diff
@@ -155,6 +160,8 @@ public String toString() {
     private final ClusterControlManager clusterControl;
     private final TimelineHashMap<String, Uuid> topicsByName;
     private final TimelineHashMap<Uuid, TopicControlInfo> topics;
+    private final TimelineHashMap<Integer, List<TopicControlInfo>> topicsByBroker;
```
Comment (aloknnikhil) [inline metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java:163]: This is a placeholder. Will be replaced with the other PR that you mentioned @cmccabe

File: metadata/src/main/resources/common/metadata/UnregisterBrokerRecord.json [hunk 2]
```diff
@@ -19,9 +19,9 @@
   "name": "UnregisterBrokerRecord",
   "validVersions": "0",
   "fields": [
-    { "name": "BrokerId", "type": "int32", "versions": "0+",
```
Comment (cmccabe) [inline metadata/src/main/resources/common/metadata/UnregisterBrokerRecord.json:22]: Previously, we agreed to use BrokerId here for consistency with the other records.

**Comment by cmccabe** [comment]
Thanks for this.

I don't think we should overload BrokerLifecycleManager#beginShutdown to handle controlled shutdown.  We should have a separate function for entering that state

In general we need to shutdown the broker lifecycle manager even if controlled shutdown is not enabled (we need to join the thread, and etc.)  same thing in KafkaClusterTestKit ... we want to be calling Kip500Broker#shutdown always, even if controlled shutdown is not enabled.  The config will dictate whether controlled shutdown happens or not.  By default it should be off in tests, to save time.

**Comment by cmccabe** [comment]
See https://github.com/confluentinc/kafka/pull/485

**Comment by cmccabe** [comment]
I reorganized this a bit and pushed it.  thanks, @aloknnikhil !

**Pull request closed without merging**